cmd/kube-controller-manager/app/names/controller_names.go

```diff
@@ -0,0 +1,108 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package names
+
+import cpnames "k8s.io/cloud-provider/app/names"
+
+// canonical controller names
+//
+// naming should be consistent
+// 1. use of shortcuts should be avoided
+// 2. kubernetes resources should be written together without a dash ("-")
+const (
+	ServiceAccountTokenControllerName                = "serviceaccount-token-controller"
```

Comment:
nit: is it stuttery to end every constant with `Name` when we're already in a package called `names`?

can we do anything in a unit test to enforce consistency on these names (all lowercase, `-` instead of `_`, end in `-controller`, etc)?

Reply:
+1 for both

I have updated `controllermanager_test` to enforce the consistency via a regex. Should I also limit the length of the controller name as we do in DNS1123 validations? Not sure if that is useful.